Plugin/Model/Customer/DataProviderPlugin.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * @category    ClassyLlama
+ * @copyright   Copyright (c) 2018 Classy Llama Studios, LLC
+ * @author      sean.templeton
+ */
+
+namespace ClassyLlama\AvaTax\Plugin\Model\Customer;
+
+use ClassyLlama\AvaTax\Exception\AvataxConnectionException;
+use ClassyLlama\AvaTax\Helper\UrlSigner;
+use Magento\Framework\DataObject;
+use Magento\Framework\DataObjectFactory;
+
+class DataProviderPlugin
+{
+
+    /**
+     * @var \ClassyLlama\AvaTax\Api\RestCustomerInterface
+     */
+    protected $customerRest;
+
+    /**
+     * @var DataObjectFactory
+     */
+    protected $dataObjectFactory;
+
+    /**
+     * @var UrlSigner
+     */
+    protected $urlSigner;
+
+    /**
+     * @var \Magento\Framework\UrlInterface
+     */
+    protected $urlBuilder;
+
+    /**
+     * @param \ClassyLlama\AvaTax\Api\RestCustomerInterface  $customerRest
+     * @param DataObjectFactory                              $dataObjectFactory
+     * @param UrlSigner                                      $urlSigner
+     * @param \Magento\Framework\UrlInterface                $urlBuilder
+     */
+    public function __construct(
+        \ClassyLlama\AvaTax\Api\RestCustomerInterface $customerRest,
+        DataObjectFactory $dataObjectFactory,
+        UrlSigner $urlSigner,
+        \Magento\Framework\UrlInterface $urlBuilder
+    )
+    {
+        $this->customerRest = $customerRest;
+        $this->dataObjectFactory = $dataObjectFactory;
+        $this->urlSigner = $urlSigner;
+        $this->urlBuilder = $urlBuilder;
+    }
+
+    /**
+     * @param $certificateId
+     *
+     * @param $customerId
+     *
+     * @return string
+     */
+    public function getCertificateUrl($certificateId, $customerId)
+    {
+        $parameters = [
+            'certificate_id' => $certificateId,
+            'customer_id' => $customerId,
+            'expires' => time() + (60 * 60 * 24) // 24 hour access
```

Comment:
Can you move this to a constant and provide more context in your comment for why we chose 24 hours?